apps/taskman/service.py

```diff
@@ -107,10 +126,43 @@ def create_or_update_task(self, flaw: Flaw, fail_if_exists=False) -> Response:
         except JIRAError as e:
             return Response(data=e.response.json(), status=e.status_code)
 
-    def update_task_status(self, issue_key: str, status: TaskStatus) -> Response:
+    def update_task_status(
+        self,
+        issue_key: str,
+        status: TaskStatus,
+        resolution: Optional[TaskResolution] = None,
+    ) -> Response:
         """Transition a task to a new state"""
         try:
-            self.jira_conn.transition_issue(issue=issue_key, transition=status)
+            if status == TaskStatus.CLOSED and resolution == TaskResolution.WONT_DO:
+                issue = self.jira_conn.issue(issue_key)
+                uuid_labels = [
+                    label for label in issue.fields.labels if "flawuuid:" in label
+                ]
+                flaw_uuid = uuid_labels[0].split(":")[1]
```

Comment:
I thinking about a case when someone manually removes the label in Jira. That should be rare and would be a human error but in such case we would end up with an **IndexError** which would not tell us much about what is actually the issue. Maybe we should check that the label was actually found and raise a concrete error if not?

Reply:
I hope that a situation when someone adds a label with non-existing UUID is too imaginary :smile:

Reply:
Makes sense, I will hope for the best and expect for the worse and raise an error :smile:

Reply:
I have added a 500 with error description for this case.

Reply:
What about 409? It's called in a `PUT` anyway so I feel like it should be OK, 500 implies server mistake where in this case we're assuming that someone or something other than us removed the label, or the passed-in key is incorrect

Reply:
Makes sense, changed for 409.